### ISSUE TITLE
azurerm_storage_account_network_rules doesn't accept CIDR notation

### ISSUE THREAD

**Issue by bruceharrison1984** [terraform-providers/terraform-provider-azurerm#5164]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Issue

- Terraform: 0.12.16
- AzureRM Provider: 1.38.0

Trying to use `azurerm_storage_account_network_rules` with a CIDR notation fails (X.X.X.X/XX) with an IP validation error.

```terraform
resource "azurerm_storage_account_network_rules" "mdw" {
  resource_group_name  = azurerm_resource_group.mdw.name
  storage_account_name = azurerm_storage_account.mdw.name

  default_action = "Deny"
  ip_rules       = ["128.252.0.0/16"]
  bypass         = ["AzureServices"]
}
```
```sh
"ip_rules.0" is not a valid IPv4 address: "128.252.0.0/16"
```

The `network_rules` section of `azurerm_storage_account` accepts both IPs and CIDR notation. Looking at the code, it appears that the `azurerm_storage_account_network_rules` makes use of validation that `azurerm_storage_account` does not.

https://github.com/terraform-providers/terraform-provider-azurerm/blob/48d259b4894f91f7485644855a47e791264f4007/azurerm/resource_arm_storage_account_network_rules.go#L66-L76

https://github.com/terraform-providers/terraform-provider-azurerm/blob/48d259b4894f91f7485644855a47e791264f4007/azurerm/resource_arm_storage_account.go#L206-L212

I would expect the behavior between these two resources to be identical. Not being able to utilize `azurerm_storage_account_network_rules` prevents a bunch of different scenarios from being possible due to circular dependencies occuring between storage and other resources.

When configuring the firewall through the Azure Portal, both CIDR and IPs are valid.

**Comment by phires** [comment]
Seems to be fixed by PR #5201, although maybe we could create a combined validation for CIDR or single IP?

**Comment by tombuildsstuff** [comment]
@phires agreed this is fixed by #5201 - thanks for the heads up - I'grll close this out

> although maybe we could create a combined validation for CIDR or single IP?

Agreed, that'd be a great addition 👍
